### ISSUE TITLE
[bug?] no-leaked-conditional-rendering: object should be considered as valid left-hand type

### ISSUE THREAD

**Issue by latin-1** [Rel1cx/eslint-react#568]
### Describe the bug

I don't see an obvious reason to ban this type.

### Reproduction

```tsx
const a = {} as {};
const b = {} as {} | null;
const b = {} as {} | undefined;
<>{a && <div />}</>
<>{b && <div />}</>
<>{c && <div />}</>
```

### Platform and versions

```Text
@eslint-react: 1.5.15-beta.3
```
